lab_6_pipeline/pipeline.py

```diff
@@ -18,22 +36,55 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self._storage = {}
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._validate_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
 
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        if not self.path_to_raw_txt_data.iterdir():
+            raise EmptyDirectoryError
+
+
+        texts = list(self.path_to_raw_txt_data.glob('*.txt'))
+        meta_info = list(self.path_to_raw_txt_data.glob('*.json'))
+        right_texts = sorted([int(re.match(r'\d+', i.name).group()) for i in texts])
+
+        if len(right_texts) != len(meta_info):
+            raise InconsistentDatasetError
+
+        for i in texts:
+            if not i.stat().st_size:
+                raise InconsistentDatasetError
+            if sorted([int(re.match(r'\d+', i.name).group()) for i in texts]) \
+                    != list(i for i in range(1, len(texts)+1)):
+                raise InconsistentDatasetError
+
     def _scan_dataset(self) -> None:
         """
         Register each dataset entry
         """
+        for file in (list(self.path_to_raw_txt_data.glob('*.txt'))):
+            num = int(re.match(r'\d+', file.name).group())
```

Comment:
to this moment we already consider corpus to be validated, no validation is required